processmanager/process_impl.go

```diff
@@ -12,57 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package process
+package processmanager
 
 import (
+	"io"
 	"os"
+	"os/exec"
 	"strings"
 	"syscall"
 	"time"
 
-	"github.com/ligato/cn-infra/process/status"
+	"github.com/ligato/cn-infra/processmanager/status"
 	"github.com/pkg/errors"
 )
 
 // Marked defines that the process should be always restarted
 const infiniteRestarts = -1
 
-func (p *Process) startProcess() (*os.Process, error) {
-	wd, err := os.Getwd()
+func (p *Process) startProcess() (cmd *exec.Cmd, err error) {
+	// set directory
+	cmd = &exec.Cmd{Path: p.cmd}
+	cmd.Dir, err = os.Getwd()
 	if err != nil {
 		return nil, errors.Errorf("failed to get rooted path name for: %v", err)
 	}
-	var attr = os.ProcAttr{
-		Dir:   wd,
-		Env:   os.Environ(),
-		Files: []*os.File{os.Stdin, nil, nil},
-	}
-	// Syscall if process should be detached from parent
-	if p.options != nil && p.options.detach {
-		attr.Sys = &syscall.SysProcAttr{
-			Setpgid: true,
-			Pgid:    0,
+
+	// if options are set, adjust command attributes, otherwise set last required fields to prepare the command
+	if p.options == nil {
+		cmd.Args = append([]string{p.cmd})
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Pdeathsig: syscall.SIGKILL,
+		}
+	} else {
+		// args
+		cmd.Args = append([]string{p.cmd}, p.options.args...)
+		// writer
+		if p.options.outWriter != nil {
+			stdout, err := cmd.StdoutPipe()
+			if err != nil {
+				p.log.Errorf("failed to get stdout pipe: %v", err)
+			}
+			p.watchOutput(p.options.outWriter, stdout)
+		}
+		if p.options.errWriter != nil {
+			errOut, err := cmd.StderrPipe()
+			if err != nil {
+				p.log.Errorf("failed to get stderr pipe: %v", err)
+			}
+			p.watchOutput(p.options.errWriter, errOut)
+		}
+		// detach
+		if p.options.detach {
```

Comment:
Shouldn't this include else case with setting of `Pdeathsig`? The **detached process** is _opt-in_ feature. To me, this implicates that we want all **non-detached processed** die with their parent process on crash/panic/kill.